langchain/vectorstores/rocksetdb.py

```diff
@@ -0,0 +1,308 @@
+"""Wrapper around Rockset vector database."""
+from __future__ import annotations
+
+import logging
+from enum import Enum
+from typing import Any, Iterable, List, Optional, Tuple
+
+# Make sure you have rockset client installed. If not, you can
+# install it with `pip install rockset`.
```

Comment:
Is there a reason we don't want to add the rockset client to the list of dependencies?

Reply:
Ok, I found this: https://github.com/anubhav94N/langchain/blob/master/.github/CONTRIBUTING.md#working-with-optional-dependencies

So unless we can get permission to make this a required dependency, this needs to at least run without errors when imported. We can put the rockset imports in try/catch statements and set a var that holds value of whether it was successfully imported, then check that var later and throw. Or you can do what they did here which is to import in the function: https://github.com/anubhav94N/langchain/blob/master/.github/CONTRIBUTING.md#working-with-optional-dependencies